rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -824,13 +812,7 @@ rmw_destroy_publisher(rmw_node_t * node, rmw_publisher_t * publisher)
         i + 1, \
         array_size, \
         values); \
-      if (values) { \
-        /* First call the destructor for each element, then free the whole block. */ \
-        for (size_t i = 0; i < array_size; ++i) { \
-          RMW_TRY_DESTRUCTOR(values[i].~DDS_TYPE(), DDS_TYPE, ) \
-        } \
-        rmw_free(values); \
-      } \
+      rmw_free(values); \
```

Comment:
I guess there is an assumption here that `rmw_free` behaves like `free` when it is given a nullptr. We'll just have to remember to do the same if we provide an alternative allocation strategy.